run.php

```diff
@@ -4,13 +4,24 @@
 
 /** @var $this \Icinga\Application\Modules\Module */
 
+use Icinga\Application\Icinga;
+use Icinga\Authentication\Auth;
+
 $this->provideHook('ApplicationState');
 $this->provideHook('X509/Sni');
 $this->provideHook('health', 'IcingaHealth');
 $this->provideHook('health', 'RedisHealth');
 $this->provideHook('Reporting/Report', 'Reporting/HostSlaReport');
 $this->provideHook('Reporting/Report', 'Reporting/ServiceSlaReport');
 
+if (
+    $this::exists('reporting') && Icinga::app()->getModuleManager()->hasEnabled('reporting')
+    && Auth::getInstance()->hasPermission('reporting/reports')
```

Comment:
Accessing the authentication at this stage (run.php runs very early) may cause issues, especially on routes that normally don't need authentication. Placing this here causes everything to require authentication as run.php runs always. Please check the permission in the hooks itself.